### PR TITLE
Modernize ESProducers in CalibTracker/SiPixelESProducers

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

This PR modernizes the ESProducers in CalibTracker/SiPixelESProducers (as part of #26748) by

- moving contents of the headers to the corresponding source files
- using ESGetToken (except for SiPixelGenErrorDBObjectESProducer and SiPixelTemplateDBObjectESProducer that have a non-trivial aspect in them and are dealt separately in #27043)
- reducing object copies in PixelFedChannelCollectionProducer

#### PR validation:

Code compiles, no changes expected.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

File: CalibTracker/SiPixelESProducers/plugins/SiPixelQualityESProducer.cc [hunk 1]
```diff
@@ -16,36 +16,86 @@
...
+  auto toGet = conf_.getParameter<std::vector< edm::ParameterSet >>("ListOfRecordToMerge");
+  auto throwIfNotFound = [&toGet](const std::string& recordName) {
+    if(std::find_if(toGet.begin(), toGet.end(), [&recordName](const edm::ParameterSet& pset) {
+          return pset.getParameter<std::string>("record") == recordName;
+        }) == toGet.end()) {
+      throw cms::Exception("Configuration") << "Did not find a PSet with record='" << recordName << "' from ListOfRecordToMerge";
+    }
+  };
+  throwIfNotFound("SiPixelDetVOffRcd");
+  throwIfNotFound("SiPixelQualityFromDbRcd");
```
Comment (makortel) [inline CalibTracker/SiPixelESProducers/plugins/SiPixelQualityESProducer.cc]: I'll be happy to remove these checks if someone could confirm that the `ListOfRecordToMerge` parameter indeed has no real purpose (anymore?).

@tsusa @mmusich

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by pohsun** [comment]
please test

File: CalibTracker/SiPixelESProducers/plugins/SiPixelQualityESProducer.cc (hunk 1 above)
Comment (tsusa) [reply CalibTracker/SiPixelESProducers/plugins/SiPixelQualityESProducer.cc]: @makortel  Sorry for a slow response. The ListOfRecordToMerge parameter has no real purpose anymore [for SiPixelDetVOffRcd the dummy payload is used].

**Comment by pohsun** [comment]
+1

**Comment by fabiocos** [comment]
+1

File: CalibTracker/SiPixelESProducers/plugins/SiPixelQualityESProducer.cc (hunk 1 above)
Comment (makortel) [reply CalibTracker/SiPixelESProducers/plugins/SiPixelQualityESProducer.cc]: Thanks @tsusa. I created #27087 to remove these checks and the parameter from the `cfi` file.

Comment (tsusa) [reply CalibTracker/SiPixelESProducers/plugins/SiPixelQualityESProducer.cc]: @makortel tnx!

**Pull request merged**